### PR TITLE
Enforce code rule IDE0032

### PR DESCRIPTION
**Pull request by RoosterDragon**
Enforces IDE0032 Use auto-implemented property
https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0032

After the fixer ran, I adjusting the location of some properties in the files to more natural locations, however there may be more tweaks we want to make here.

Additionally in `Target.cs` it felt like we didn't want to make the fields "public" via the serialization stuff. So I introduced a `SerializableState` method so that we can keep private fields without the rule wanting to turn them into auto-props.

### CODE REVIEW

**Comment by penev92** [comment]
Needs a rebase.

File: OpenRA.Game/FileSystem/Folder.cs [hunk 1]
```diff
@@ -18,39 +18,37 @@ namespace OpenRA.FileSystem
 {
 	public sealed class Folder : IReadWritePackage
 	{
-		readonly string path;
+		public string Name { get; }
```
Comment (abcdefg30) [inline OpenRA.Game/FileSystem/Folder.cs:21]: Bit of a scope creep, but why did we expose that as `Name` and not `Path`? (And now you won't even be able to tell be looking at the previous "getter".)

File: OpenRA.Game/FileSystem/ZipFile.cs [hunk 2]
```diff
@@ -142,23 +142,22 @@ public void Delete(string filename)
 
 		sealed class ZipFolder : IReadOnlyPackage
 		{
-			public string Name => path;
+			public string Name { get; }
```
Comment (abcdefg30) [inline OpenRA.Game/FileSystem/ZipFile.cs:145]: Same here.

File: OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs [hunk 3]
```diff
@@ -142,7 +141,7 @@ public class ProductionQueue : IResolveOrder, ITick, ITechTreeElement, INotifyOw
 		protected DeveloperMode developerMode;
 		protected TechTree techTree;
 
-		public Actor Actor => self;
+		public Actor Actor { get; }
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs:144]: Fwiw this does feel a bit weird since we use `self` everywhere... but I guess this is unavoidable here.

File: OpenRA.Game/FileSystem/Folder.cs (hunk 1 above)
Comment (RoosterDragon) [reply OpenRA.Game/FileSystem/Folder.cs:21]: It's `Name` to implement `IReadOnlyPackage.Name`. Do we want to rename the interface and all implementations to `Path`? That feels like a larger scope change than you may have intended with your suggestion.

**Comment by PunkPun** [comment]
I don't think I agree with this PR at its current state. I think every variable name needs a review, and I also recon some variables may want to become `public readonly` instead of `public { get; }`

File: OpenRA.Game/FileSystem/Folder.cs (hunk 1 above)
Comment (abcdefg30) [reply OpenRA.Game/FileSystem/Folder.cs:21]: Yeah it is probably best left for another PR if we want to change this. `Name` just implies to me that this is the package name, not the entire path.

File: OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs (hunk 3 above)
Comment (RoosterDragon) [reply OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs:144]: We have a few options for the case where the property and backing field use different names:

1. Turn on the rule and the lost fields names in these few places become acceptable losses for the benefit of having the rule. An alternative point of view is that the rule reduces confusion, because you can't have a field and prop with mismatching names.
2. Turn on the rule, but disable it in places where we want to keep different names. This keeps the names but means future cases would need to know to disable the rule if they also wanted it which might be non-obvious.
3. Don't enable this rule, as we think this cost isn't worth the other benefits.

File: OpenRA.Game/Graphics/UISpriteRenderable.cs [hunk 4]
```diff
@@ -39,26 +36,26 @@ public UISpriteRenderable(Sprite sprite, WPos effectiveWorldPos, int2 screenPos,
 			// HACK: This is working around the fact that palettes are defined on traits rather than sequences
 			// and can be removed once this has been fixed
 			if (sprite.Channel == TextureChannel.RGBA && !(palette?.HasColorShift ?? false))
-				this.palette = null;
+				Palette = null;
 		}
 
 		// Does not exist in the world, so a world positions don't make sense
-		public WPos Pos => effectiveWorldPos;
+		public WPos Pos { get; }
```
Comment (PunkPun) [inline OpenRA.Game/Graphics/UISpriteRenderable.cs:43]: This comment should be moved. It doesn't make sense to have it at an interface implementation.

It also may make sense to change the `IRenderable.Pos`name into `IRenderable.EffectiveWorldPos`. It's called like this in loads of places. But I haven't touched that part of the code so I dunno if this suggestion makes sense.

Incase we do go along with this, this commend can become a remark

File: OpenRA.Mods.Common/Traits/Armament.cs [hunk 5]
```diff
@@ -136,7 +134,7 @@ public class Armament : PausableConditionalTrait<ArmamentInfo>, ITick
 		public Armament(Actor self, ArmamentInfo info)
 			: base(info)
 		{
-			this.self = self;
+			Actor = self;
```
Comment (PunkPun) [inline OpenRA.Mods.Common/Traits/Armament.cs:137]: In the future we may want to consider refactoring `AttackGarrisoned` to remove this

File: OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs (hunk 3 above)
Comment (PunkPun) [reply OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs:144]: IMO this is fine. Fixing this properly would require rewriting production code, which is something we already want to do, and which is also out of scope of this PR

File: OpenRA.Game/Graphics/UISpriteRenderable.cs (hunk 4 above)
Comment (PunkPun) [reply OpenRA.Game/Graphics/UISpriteRenderable.cs:43]: More investigating shows that Pos, is an actual world position in some implementations.

Additionally, I found is that `IRenderable.Pos` only exists to be used here

https://github.com/OpenRA/OpenRA/blob/17a2c6a3832221982c7a4b4f3f4b934d850d32ad/OpenRA.Game/Graphics/WorldRenderer.cs#L23-L24

Comment (RoosterDragon) [reply OpenRA.Game/Graphics/UISpriteRenderable.cs:43]: The use case is to sort the renderables in the world bottom to top so units and such render above each other correctly. Some renderables such as UI elements don't have the notion of being "in the world", which is where the "effective world pos" comes from. The current naming and comments make sense in this context.

Comment (PunkPun) [reply OpenRA.Game/Graphics/UISpriteRenderable.cs:43]: Yes, that's what I gathered. This really seems like bad naming. I'm not sure how to go forward with this, I can think of multiple solutions but none seem great

* Add a summary to `IRenderable.Pos` that it's only supposed to be used for calculating Z comparison key
   * This just makes it less confusing, but keeps the shuehorn implementation
* Rename `IRenderable.Pos` to `IRenderable.EffectivePos`
   * Downside is that where it defines real position it becomes misleading
* Refactor so that every `IRenderable` would have to define it's Z comparison key, thus removing `IRenderable.Pos`
   * Downside is that this causes a lot of duplication

Comment (RoosterDragon) [reply OpenRA.Game/Graphics/UISpriteRenderable.cs:43]: Since the naming is pre-existing, it doesn't block this PR at least.

Comment (PunkPun) [reply OpenRA.Game/Graphics/UISpriteRenderable.cs:43]: But we do rename a lot of variables here

Comment (RoosterDragon) [reply OpenRA.Game/Graphics/UISpriteRenderable.cs:43]: All the existing properties retain their names. The only "renaming" is when the backing fields disappear - and some have different names to the publically exposed property.

**Review by abcdefg30 — APPROVED** [review]
I guess we can defer renaming to a follow-up PR.

File: OpenRA.Game/Graphics/Viewport.cs [hunk 6]
```diff
@@ -86,12 +83,13 @@ private set
 			}
 		}
 
-		public float MinZoom => minZoom;
+		public float MinZoom { get; private set; } = 1f;
+		public float MaxZoom { get; private set; } = 2f;
```
Comment (penev92) [inline OpenRA.Game/Graphics/Viewport.cs:87]: This exposes something that wasn't exposed before but I think that's fine.

**Review by penev92 — APPROVED** [review]
The Folder.path -> Folder.Name change will be confusing, that's for sure, because we're dealing with the full folder path, not just its name.

There are a few things that don't look as good/clear, but this is what I get for pushing for enforcing more rules.

**Comment by PunkPun** [comment]
> Folder.path -> Folder.Name

That's a problem of the file system itself. That's just how paths are named

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/43b288724a4ec96f5c9eaec31c05a2fbe29143a3...83bd4ae680e9da4ca1874ffa5a728ac25c6b171c)

File: OpenRA.Game/Graphics/Viewport.cs (hunk 6 above)
Comment (RoosterDragon) [reply OpenRA.Game/Graphics/Viewport.cs:87]: Yes, it seemed sensible to have both the min and max zoom available so I decided to make that change manually.
